conda_smithy/lint_recipe.py

```diff
@@ -149,6 +151,15 @@ def lintify(meta, recipe_dir=None):
     except RuntimeError as e:
         lints.append(str(e))
 
+    # 13: Subheaders should be in the allowed subheadings
+    for section in major_sections:
+        expected_subsections = FIELDS[section]
```

Comment:
Are we certain that a section exists in FIELDS? If not, we should be a bit more defensive about getting hold of the section fields... ```FIELDS.get(section, [])``` would do.